src/mz/src/command/region.rs

```diff
@@ -50,24 +50,33 @@ pub async fn enable(cx: RegionContext, version: Option<String>) -> Result<(), Er
         .max_duration(Duration::from_secs(720))
         .clamp_backoff(Duration::from_secs(1))
         .retry_async(|_| async {
-            let region_info = cx.get_region().await?.region_info;
+            let region = cx.get_region().await?;
 
-            match region_info {
-                Some(region_info) => {
+            match region.region_state {
+                RegionState::EnablementPending => {
                     loading_spinner.set_message("Waiting for the region to be ready...");
-                    if region_info.resolvable {
-                        if cx
-                            .sql_client()
-                            .is_ready(&region_info, cx.admin_client().claims().await?.email)?
-                        {
-                            return Ok(());
+                    Err(Error::NotReadyRegion)
+                }
+                RegionState::DeletionPending | RegionState::SoftDeleted => Err(
+                    Error::CommandExecutionError("Error enabling region".to_string()),
```

Comment:
I wonder if we want to be more explicit for these cases? "This region has been scheduled for deletion," or something similar?

Reply:
+1. We should. It helps to debug faster and provide more context/value to the user.